File: Data Structures & Algorithms/max-water-container/submission-2.cs
public class Solution {
    public int MaxArea(int[] heights) {
        int l = 0, r = heights.Length-1, max_height = 0;
        while (l < r) {
            int height = (r-l) * Math.Min(heights[l], heights[r]);
            max_height = Math.Max(max_height, height);

            if (heights[l] < heights[r]) {
                l++;
            } else {
                r--;
            }
        }
        return max_height;
    }
}

File: Data Structures & Algorithms/coin-change/submission-2.cpp
class Solution {
public:
    int coinChange(vector<int>& coins, int amount) {
        vector<int> dp(amount+1, amount+1);
        dp[0] = 0;

        for (int a = 1; a < amount+1; ++a) {
            for (int c : coins) {
                if (a-c >= 0) {
                    dp[a] = min(dp[a], 1 + dp[a-c]);
                }
            }
        } 
        return dp[amount] != amount + 1 ? dp[amount] : -1;
    }
};
